src/file/metadata.rs

```diff
@@ -354,8 +374,391 @@ impl ColumnChunkMetaData {
       total_uncompressed_size,
       data_page_offset,
       index_page_offset,
-      dictionary_page_offset
+      dictionary_page_offset,
+      statistics
     };
     Ok(result)
   }
 }
+
+/// Column order to compare values with statistics min/max.
+#[derive(Clone, Debug, PartialEq)]
+pub enum ColumnOrder {
+  /// Signed (either value or legacy byte-wise) comparison.
+  SIGNED,
+  /// Unsigned (depending on physical type either value or byte-wise) comparison.
+  UNSIGNED,
+  /// Comparison is undefined.
+  UNDEFINED
+}
+
+/// Optional column statistics that are used per row group and per page.
+///
+/// Use `order` field to determine comparison order for a type.
+/// When statistics contain deprecated min/max fields, then comparison is always signed.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Statistics {
+  Boolean { order: ColumnOrder, min: Option<bool>, max: Option<bool>, nulls: u64 },
```

Comment:
wonder if it would be better to have a `Statistics` trait with methods such as `null_count`, `has_nulls`, etc., and a separate struct `TypedStatistics<T>` that implements the former. The latter can implement more methods in future such as `set_min_max`, etc., that are specific to the generic type `T`.

Reply:
Sorry, I am a bit confused. I chose not using `T`, so it is easy to extract relevant statistics a type.

My main concern with typed statistics is that we would need a wrapper similar to column reader. That is why I chose to remove this part and directly implement as enum of different statistics for physical type.

Could you elaborate a little bit more your plan on adding typed statistics? Should we consider statistics as read-only? Thanks.

Reply:
Yes I was wondering if this could be similar to the `RowGroupStatistics` in the C++ version: https://github.com/apache/parquet-cpp/blob/master/src/parquet/statistics.h#L83, which is readonly, and then TypedStatistics for different types. 

I'm not very familiar with how statistics will be used in the whole codebase though. I think the current approach is fine as well - we can come back and revisit this if needed.

Reply:
Interesting... let me think about this. I do want to expose min/max methods.